tutorials/external-metastore-for-dataproc/index.md

```diff
@@ -0,0 +1,278 @@
+        --region ${REGION} \
+        --zone ${ZONE} \
+        --single-node \
+        --master-machine-type n1-standard-8 \
+        --master-boot-disk-type pd-ssd \
+        --master-boot-disk-size 500 \
+        --image-version 2.0.29-debian10 \
+        --properties hive:hive.metastore.warehouse.dir=gs://${PROJECT}-warehouse/datasets \
+        --initialization-actions gs://goog-dataproc-initialization-actions-${REGION}/cloud-sql-proxy/cloud-sql-proxy.sh \
+        --metadata hive-metastore-instance=${PROJECT}:${REGION}:hive-metastore-db
+
+This command will take a few minutes to create and initialize the cluster.
+
+Create a failover metastore service cluster using the following command in Cloud Shell:
+
+        gcloud dataproc clusters create hive-metastore2 \
+        --async
+        --enable-component-gateway \
+        --scopes sql-admin \
+        --region ${REGION} \
+        --zone ${ZONE} \
+        --single-node \
+        --master-machine-type n1-standard-8 \
+        --master-boot-disk-type pd-ssd \
+        --master-boot-disk-size 500 \
+        --image-version 2.0.29-debian10 \
+        --properties hive:hive.metastore.warehouse.dir=gs://${PROJECT}-warehouse/datasets \
+        --initialization-actions gs://goog-dataproc-initialization-actions-${REGION}/cloud-sql-proxy/cloud-sql-proxy.sh \
+        --metadata hive-metastore-instance=${PROJECT}:${REGION}:hive-metastore-db
+
+### Create a Hive table
+
+In this section, you upload a sample dataset to your warehouse bucket, create a new external Hive table.
+
+In Cloud Shell run the following commands:
+
+1.  Copy the sample dataset to warehouse bucket:
+
+        gsutil cp gs://hive-solution/part-00000.parquet \
+        gs://${PROJECT}-warehouse/datasets/transactions/part-00000.parquet
+
+    The sample dataset is compressed in the [Parquet](https://parquet.apache.org/) format and contains thousands of fictitious bank transaction records with three columns: date, amount, and transaction type.
+
+1.  Create an external Hive table for the dataset:
+
+        gcloud dataproc jobs submit hive \
+        --cluster hive-metastore1 \
+        --region ${REGION} \
+        --execute " 
+        CREATE EXTERNAL TABLE transactions (SubmissionDate DATE, TransactionAmount DOUBLE, TransactionType STRING) STORED AS PARQUET 
+        LOCATION 'gs://${PROJECT}-warehouse/datasets/transactions';"
+
+### Create Worker clusters
+
+In this section, you create two Dataproc clusters in different regions preferrably in the same multi-region covered by the warehouse bucket.
+
+In Cloud Shell, run the following command to create first worker cluster in the same region:
+
+        gcloud dataproc clusters create hive-worker1 \
+        --image-version 2.0.29-debian10 \
+        --region ${REGION} \
+        --properties hive:hive.metastore.warehouse.dir=gs://${PROJECT}-warehouse/datasets \
+        --properties hive:hive.metastore.uris=thrift://hive-metastore1-m:9083,thrift://hive-metastore2-m:9083
```

Comment:
sorry, I don't understand what the `^!^` is doing?

Reply:
Since there is a comma in the property value, the delimiter needs to be set to something other than a comma:
https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/cluster-properties#formatting

Reply:
added in the latest commit